internal/kafka/internal/workers/kafka_mgrs/promotion/chain/reconciler_chain.go

```diff
@@ -0,0 +1,69 @@
+package chain
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/dbapi"
+)
+
+// ReconcileAction - This is the interface that must be implemented by each action that the reconciler will execute.
+// T - is the type of result the action will generate
+type ReconcileAction[T any] interface {
+	// PerformJob - This is the method where the Action implementation will perform the real job
+	// * kafkaRequest - the kafka request to reconcile
+	// * currentResult - object containing the results that have been produced up to this point into the pipeline
+	// Return values
+	// * res Result[T] - the result produced by this action
+	// * finished bool - returning `true` here will interrupt the pipeline execution
+	// * err error - an eventual error. Returning an error always interrupts the pipeline execution
+	PerformJob(kafkaRequest *dbapi.KafkaRequest, currentResult Result[T]) (res Result[T], finished bool, err error)
+}
+
+// Result - This struct contains the result produced by each action
+type Result[T any] struct {
+	hasValue bool
+	value    T
+}
+
+func (r *Result[T]) SetValue(value T) {
+	r.value = value
+	r.hasValue = true
+}
+
+func (r *Result[T]) Value() T {
+	return r.value
+}
+
+func (r *Result[T]) HasValue() bool {
+	return r.hasValue
+}
+
+// The ReconcileActionRunner is the engine of the action chain. It runs each action one after the other
+// passing to each of them the current result so that all the action can concur to create a piece of the
+// result. When the chain execution is finished, the result will be complete.
+type ReconcileActionRunner[T any] struct {
+	actions []ReconcileAction[T]
+}
+
+func NewReconcileActionRunner[T any](actions ...ReconcileAction[T]) ReconcileActionRunner[T] {
+	return ReconcileActionRunner[T]{
+		actions: actions,
+	}
+}
+
+// Run the chain end return the final result of the execution
```

Comment:
Let's document the behavior of how the intermediate results are managed and also the return value. By that I mean: are intermediate results merged? overwritten? is the final result a merge of all the results? the result of the latest executed action?